### PR TITLE
Domain init cleanup

### PR DESCRIPTION
**Pull request by 0leks**
Bunch of non-breaking cleanup that doesn't change functionality.

TODO run the image use cases and verify that they still work compared to master branch
- ~ellipsoid~
- ~ellipsoid_fd~
- ~left_atrium~
- ~femur~

### CODE REVIEW

**Comment by akenmorris** [comment]
When dealing with changing the order of initialization, we will have to check examples like "fixed_domains" and stuff like that.

File: Libs/Optimize/Optimize.cpp [hunk 1]
```diff
@@ -228,6 +228,24 @@ int Optimize::GetDomainsPerShape()
   return this->m_domains_per_shape;
 }
 
+//---------------------------------------------------------------------------
+void Optimize::SetDomainType(std::string& domain_type)
```
Comment (akenmorris) [inline Libs/Optimize/Optimize.cpp:232]: Should we use an enum or something rather than a std::string?

File: Libs/Optimize/Optimize.h [hunk 2]
```diff
@@ -81,6 +82,9 @@ class Optimize
   //! Return the number of domains per shape
   int GetDomainsPerShape();
 
+  void SetDomainType(std::string &_domain_type);
```
Comment (akenmorris) [inline Libs/Optimize/Optimize.h:85]: Is that a leading underscore?

File: Libs/Optimize/Optimize.cpp (hunk 1 above)
Comment (0leks) [reply Libs/Optimize/Optimize.cpp:232]: This is the string it loads from the params file. At every point after this it uses the DomainType enum.

Comment (akenmorris) [reply Libs/Optimize/Optimize.cpp:232]: Oh, you have a DomainType.  So it's a question of where the conversion from the parameter file's string takes place...

Comment (akenmorris) [reply Libs/Optimize/Optimize.cpp:232]: Should we move it to OptimizeParameterFile.cpp?  I guess the mismatch of get/set is bothering me.

File: Libs/Optimize/Optimize.h (hunk 2 above)
Comment (0leks) [reply Libs/Optimize/Optimize.h]: Oops

File: Libs/Optimize/Optimize.cpp (hunk 1 above)
Comment (0leks) [reply Libs/Optimize/Optimize.cpp]: Mismatch fixed

**Comment by akenmorris** [comment]
All looks ok to me now.  Ok to merge pending the test results.  @0leks you are running LA?

**Comment by 0leks** [comment]
> All looks ok to me now. Ok to merge pending the test results. @0leks you are running LA?

Yeah Im running LA single and multiscale to compare to master branch

**Comment by 0leks** [comment]
Ill also run ellipsoid_fd after that which will be very quick

**Comment by akenmorris** [comment]
Yes, please check FD. I need to add the automated test for it.

**Comment by 0leks** [comment]
ellipsoid_fd looks good

**Comment by 0leks** [comment]
This PR does not encounter the narrow band issue for femur multiscale. It is ready to go.

**Comment by akenmorris** [comment]
@0leks, was a a different branch you were having the problem with femur?

**Comment by 0leks** [comment]
Yes the femur problem is on the mesh_domain_newest branch.

**Pull request merged**